### PR TITLE
Update/publish calypso sass

### PR DESCRIPTION
**Pull request by sirreal**
This is a spike to make 3188-gh-happy-tools possible.

This was implemented by addressing issues one at a time as they came up without much thought. It serves to understand what blockers exist, but not necessarily how they should be addressed.

### CODE REVIEW

**Comment by sirreal** [comment]
Let's wait for #36175

File: webpack.config.js [hunk 1]
```diff
@@ -212,10 +212,7 @@ const webpackConfig = {
 			SassConfig.loader( {
 				preserveCssCustomProperties: true,
 				includePaths: [ path.join( __dirname, 'client' ) ],
-				prelude: `@import '${ path.join(
-					__dirname,
-					'client/assets/stylesheets/shared/_utils.scss'
-				) }';`,
+				prelude: `@import '~@automattic/calypso-sass-prelude/sass/utils';`,
```
Comment (ockham) [inline webpack.config.js]: Should we also `packages/calypso-ui` accordingly? (See https://github.com/Automattic/jetpack/pull/13490#issuecomment-540520866)

**Comment by ockham** [comment]
@Automattic/team-calypso I'd be curious to hear your thoughts on this, esp in the context of https://github.com/Automattic/jetpack/pull/13490#issuecomment-540468074 (Does this approach make sense, is it viable?) Luna is spread a bit thin ATM, so it'd be cool to pick your brains :sweat_smile:

**Comment by jsnajdr** [comment]
There are two types of shared SCSS files that should be clearly separated:

**SCSS header/prelude files that don't output any CSS**
These are SCSS variable definitions, functions and "placeholder selectors" that are further used in SCSS rules that output actual CSS:

**Base styles that don't belong to any component, but setup the CSS environment**
- CSS reset stylesheet
- shared animations (named keyframes definitions)
- styles for form elements, as we apply Calypso styling even to plain `input` or `select` elements that don't have any class names

The current `client/assets/stylesheets` folder mixes them all together without clear distinction. That would be nice to fix first.

**The `z-index` map and function**
That's a big 🤦‍♂ that should be removed completely. There's nothing much reusable in it, as it specifies a map of Calypso-specific class names to their respective z-indexes. That map shouldn't be needed at all: instead, our HTML markup should be improved to exploit the natural z-indexing rules and stacking contexts. Described also in the [Calypso CSS guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#z-index)

**Comment by simison** [comment]
FYI, I just did a similar PR in Gutenberg: https://github.com/WordPress/gutenberg/pull/17883

>CSS reset stylesheet

This could be turned into a mixin. See [example in Gutenberg](https://github.com/WordPress/gutenberg/blob/79e5749a198b0a851d654143abf1093b51c958ed/assets/stylesheets/_mixins.scss#L385-L607).

**Comment by ockham** [comment]
Thanks for that overview, @jsnajdr!

Are those things that Team Calypso can work on in the near future? You clearly seem to have a better grasp of what's going on with those files than I do -- not sure if you also have the time to work on them? :sweat_smile:

**Comment by sirreal** [comment]
> There are two types of shared SCSS files that should be clearly separated:

💯

> **SCSS header/prelude files that don't output any CSS**
> These are SCSS variable definitions, functions and "placeholder selectors" that are further used in SCSS rules that output actual CSS

We can set clear rules here and add tooling to validate that this bit produces no output. We were recently burned by that when we started writing CSS custom properties over and over.

> **Base styles that don't belong to any component, but setup the CSS environment**
> 
> * CSS reset stylesheet
> * shared animations (named keyframes definitions)
> * styles for form elements, as we apply Calypso styling even to plain `input` or `select` elements that don't have any class names

CSS variables is another one for this list 😉

**Comment by simison** [comment]
> CSS variables is another one for this list 😉

Which bit specifically? 😁 [Performance issues?](https://github.com/Automattic/wp-calypso/pull/35949#issuecomment-531596945) (we should open an issue about that)

**Comment by sirreal** [comment]
> > CSS variables is another one for this list 😉
> 
> Which bit specifically? 😁 [Performance issues?](https://github.com/Automattic/wp-calypso/pull/35949#issuecomment-531596945) (we should open an issue about that)

CSS custom properties (like our color schemes) are part of a base style set that don't belong to any component. They belong to the environment and should be included exactly once (we had an issue recently where they were included in every component 😱 — that's the performance issue you mentioned). They're also necessary for more specific styles to work correctly.

**Pull request closed without merging**